bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,450 @@
+
+* A dev server is an hollow server scanning the directory target/deployments.
+* A dev app is the primary artifact copied to the target/deployments dir.
+
+The plugin attempts to force the provisioning of the deployment scanner in dev mode:
+
+* If deployment-scanner is excluded, remove it from exclusion. A warning message is displayed advertising that the provisioning of the deployment-scanner is enforced.
+* Add the deployment-scanner layer to the set of layers. An info message is displayed advertising that the deployment-scanner is provisioned.
+* If a provisioning.xml file is set (and no layers override it), a warning is displayed advertising that we can’t enforce the presence of the deployment-scanner.
+* If the deployment-scanner is not in the config, the CLI operation will abort the plugin execution and log an error message.
+
+===== Workflow examples
+
+Description of the 2 workflows (with and without dev mode). These examples could be designed differently according to the content of the pom file.
+
+Nominal mode, full repackaging of the server done for each rebuild, server restarted after each rebuild (not viable):
+
+* mvn package ⇒ full repackaging
+* mvn wildfly-jar:run
+* (kill synchronous execution).
+* User make changes
+* mvn package ⇒ full repackaging
+* ...
+
+Dev mode, server built/started once, app automatically re-deployed.
+
+* mvn wildfly-jar:dev ⇒ hollow server built and started
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically deployed.
+* User make changes
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically re-deployed.
+* ...
+* mvn wildfly-jar:shutdown
+* User is fine with his changes, he can do the server+app packaging: mvn package
+
+NB: This workflow doesn’t require support in IDE, it is 100% Maven. It could be optimized with IDE plugin (e.g.: netbeans plugin to track static 
+files and avoid to re-package in this case).
+
+
+==== Cloud context
+
+The plugin allows to generate a bootable JAR usable in various Cloud execution contexts: custom container, Google JIB, Fabric8, Java s2i build.
+
+As an example, s2i binary build of the https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/mp-config/README.md[microprofile-config example] is provided. 
+
+
+===== Configuring the Maven plugin for Cloud environment
+
+The Maven plugin configuration item ```<cloud></cloud>``` allows to build a bootable JAR for cloud environment. By default the server is configured to run inside an OpenShift context.
+Set the cloud child element ```<type>openshift|kubernetes</type>``` to select the targeted cloud platform.
+
+The sever configuration is updated in order to properly operate in a cloud environment:
+
+* The ```microprofile-health``` and ```core-tools``` (that contains WildFly CLI) galleon layers are provisioned. They are required for the  OpenShift probes and WildFly OpenShift operator to properly operate.
```

Comment:
So IIUC this as weel is another implicit requirement for https://issues.redhat.com/browse/WFLY-13305 to incorporate (or should actually be blocked by) https://issues.redhat.com/browse/WFLY-12681?

Reply:
microprofile-health is already present in WildFly 21.

Reply:
Yeah, true. Since `19` actually. Again I was thinking about the usage of the layer and so its related development.
Thanks.